modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/nhttp/config/ProxyConfigBuilder.java

```diff
@@ -1,82 +1,261 @@
+            log.info(name + " Proxy profiles initialized for " + proxyProfileMap.size() + " targetHosts");
+            return proxyProfileMap;
+        }
+        return null;
+    }
+
+    /**
+     * Extracts the proxy server detail from given profile
+     * @param profile profile element
+     * @param targetHosts targetHosts given in the profile
+     * @return proxyServer(HttpHost) configured in the given profile element
+     * @throws AxisFault, if host or port element is not configured properly
+     */
+    private HttpHost getHttpProxy(OMElement profile, String targetHosts) throws AxisFault {
+        String proxyHost;
+        String proxyPortStr;
+        OMElement proxyHostEle = profile.getFirstChildWithName(Q_PROXY_HOST);
+        if (proxyHostEle != null) {
+            proxyHost = proxyHostEle.getText();
+            OMElement proxyPortEle = profile.getFirstChildWithName(Q_PROXY_PORT);
+            if (proxyPortEle != null) {
+                proxyPortStr = proxyPortEle.getText();
+            } else {
+                throw new AxisFault("Proxy Port didn't configure correctly in proxy profile [" + targetHosts + "]");
             }
+        } else {
+            throw new AxisFault("Proxy Host didn't configure correctly in proxy profile [" + targetHosts + "]");
+        }
+
+        int proxyPort = Integer.parseInt(proxyPortStr);
+        return new HttpHost(proxyHost, proxyPort >= 0 ? proxyPort : 80);
+    }
+
+    /**
+     * Extracts the credential from given profile
+     * @param profile profile element
+     * @return usernamePasswordCredentials if username, password is configured, null otherwise
+     */
+    private UsernamePasswordCredentials getUsernamePasswordCredentials(OMElement profile) {
+        UsernamePasswordCredentials proxyCredentials = null;
+        OMElement proxyUserNameEle = profile.getFirstChildWithName(Q_PROXY_USER);
+        if (proxyUserNameEle != null) {
+            String proxyUserName = proxyUserNameEle.getText();
+            OMElement proxyPasswordEle = profile.getFirstChildWithName(Q_PROXY_PASSWORD);
+            String proxyPassword = proxyPasswordEle != null ? proxyPasswordEle.getText() : "";
+            proxyCredentials = new UsernamePasswordCredentials(proxyUserName,
+                    proxyPassword != null ? proxyPassword : "");
         }
-        return this;
+        return proxyCredentials;
     }
-    
-    public ProxyConfig build() {
-        return new ProxyConfig(proxy, proxycreds, proxyBypass);
+
+    /**
+     * Extracts the proxyBypass hosts from given profile
+     * @param profile profile element
+     * @return Set of String containing bypass hosts; empty set if bypass hosts is not configured
+     */
+    private Set<String> getProxyBypass(OMElement profile) {
+        Set<String> bypassSet = new HashSet<String>();
+        OMElement bypassEle = profile.getFirstChildWithName(Q_BYPASS);
+        if (bypassEle != null && !bypassEle.getText().equals("")) {
```

Comment:
What if "bypassEle.getText()" returns null? Always compare the variable value against a known constant to avoid that additional Null check required in similar instances.